Backend/Backend/Controllers/ProfileController.cs

```diff
@@ -114,4 +118,93 @@ public async Task<ActionResult> Avatar()
         // Otherwise, return the avatar
         return PhysicalFile(GetUserAvatarPath(user.Avatar), GetFileType(user.Avatar));
     }
+
+    #endregion 
+
+    #region Other Users
+
+    [HttpGet("search")]
+    public async Task<IActionResult> ProfileSearch(string searchTerm = "", int page = MIN_PAGE, int pageSize = DEFAULT_PAGE_SIZE)
+    {
+        try
+        {
+            // Identify User from JWT Token
+            User? user = await _authService.IdentifyUserAsync(HttpContext);
+
+            // If User is not found, return 404
+            if (user is null)
+                return NotFound("User does not exist.");
+
+            return Ok(await _profileService.SearchUserProfiles(searchTerm,page,pageSize,GetDomainUrl(HttpContext)));
+        }
+        catch (Exception e)
+        {
+            return BadRequest(e.Message);
+        }
+    }
+
+    [HttpGet("{userId}/get")]
+    public async Task<IActionResult> GetUser(Guid userId)
+    {
+        try
+        {
+            // Identify User from JWT Token
+            User? user = await _authService.IdentifyUserAsync(HttpContext);
+
+            // If User is not found, return 404
+            if (user is null)
+                return NotFound("User does not exist.");
+
+            return Ok(await _profileService.GetUserProfile(userId, GetDomainUrl(HttpContext)));
+        }
+        catch (Exception e)
+        {
+            return BadRequest(e.Message);
+        }
+    }
+
+    [HttpGet("{userId}/avatar")]
+    public async Task<IActionResult> GetUserAvatar(Guid userId)
+    {
+        try
+        {
+            // Identify User from JWT Token
+            User? user = await _authService.IdentifyUserAsync(HttpContext);
+
+            // If User is not found, return 404
+            if (user is null)
+                return NotFound("User does not exist.");
+
+            // Get the avatar name of the user. 
+            string avatarName = await _profileService.GetUserAvatarName(userId);
+
+            // If the avatar name is the default avatar name, return the default avatar. 
+            // Otherwise, return the user's avatar.
+            return avatarName == Models.User.DEFAULT_AVATAR_NAME ?
+                Redirect(Models.User.DEFAULT_AVATAR_URL) :
+                PhysicalFile(GetUserAvatarPath(avatarName), GetFileType(avatarName));
+        }
+        catch (Exception e)
+        {
+            return BadRequest(e.Message);
+        }
+    }
+
+    #endregion
+
+    /// <summary>
+    /// Returns the domain url of the server.
+    /// </summary>
+    /// <param name="context"></param>
+    /// <returns></returns>
+    private static string GetDomainUrl(HttpContext context)
```

Comment:
Seeing how many times we use this function, we could just add this to a helper class for controllers at this point

Reply:
Will do.